libraries/rush-lib/src/scripts/install-run.ts

```diff
@@ -169,18 +188,18 @@ function _resolvePackageVersion(
       }
 
       const npmViewVersionOutput: string = npmVersionSpawnResult.stdout.toString();
-      const versionLines: string[] = npmViewVersionOutput.split('\n').filter((line) => !!line);
-      const latestVersion: string | undefined = versionLines[versionLines.length - 1];
+      const parsedVersionOutput: string | string[] = JSON.parse(npmViewVersionOutput);
+      const versions: string[] = Array.isArray(parsedVersionOutput)
+        ? parsedVersionOutput
+        : [parsedVersionOutput];
+      versions.sort(_compareVersionStrings);
+      const latestVersion: string | undefined = versions[versions.length - 1];
```

Comment:
```suggestion
      let latestVersion: string | undefined = versions[0];
      for (let i: number = 1; i < versions.length; i++) {
        const version: string = versions[i];
        if (_compareVersionStrings(version, latestVersion) > 0) {
          latestVersion = version;
        }
      }
```

This algorithm has O(n) runtime complexity instead of O(log(n))